joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonAstVisitor.scala

```diff
@@ -252,13 +252,22 @@ class PythonAstVisitor(
    * The lowering is:
    * func = f1(arg)(f2(func))
    *
-   * This function takes a method ref, wrappes it in the decorator calls and returns the resulting expression.
+   * This function takes a method ref, wraps it in the decorator calls and returns the resulting expression.
    * In the example case this is:
    * f1(arg)(f2(func))
    */
   def wrapMethodRefWithDecorators(methodRefNode: nodes.NewNode, decoratorList: Iterable[ast.iexpr]): nodes.NewNode = {
     decoratorList.foldRight(methodRefNode)((decorator: ast.iexpr, wrappedMethodRef: nodes.NewNode) =>
-      createCall(convert(decorator), "", lineAndColOf(decorator), wrappedMethodRef :: Nil, Nil)
+      val (decoratorNode, decoratorName) = convert(decorator) match {
+        case decoratorNode: NewIdentifier =>
+          decoratorNode -> decoratorNode.name
+        case decoratorNode: NewCall =>
+          decoratorNode -> decoratorNode.name
```

Comment:
Only for identifiers giving a name makes somewhat sense. In the `NewCall` case we have no clue about the name of the returned decorator function.

Reply:
So in that case please leave the name empty.